### ISSUE TITLE
ERROR dial unix /run/user/1012/keybase/keybased.sock: connect: permission denied

### ISSUE THREAD

**Issue by jamesongithub** [keybase/keybase-issues#3090]
I installed keybase on ubuntu 16 as a non root user with sudo priveleges. Afterwards I ran `run_keybase`. I assumed this would install for all users. 

However when I try to user keybase with another user I'm getting this error: 
```
ERROR dial unix /run/user/1012/keybase/keybased.sock: connect: permission denied
```

The socket seems to be owned by the user who installed keybase. However seem grant execute to all users?
```
srwxrwxr-x 1 james james   0 Oct 17 20:38 keybased.sock
```

**Comment by jamesongithub** [comment]
```
Username:      
Logged in:     no
Session:       no session
    is valid:  no
    keys:      locked

Key status:
    stream:    not cached
    secret:    not stored
    dev sig:   not cached
    dev enc:   not cached
    paper sig: not cached
    paper enc: not cached
    prompt:    show
    tsec:      not cached

KBFS:
    status:    not running
    version:   1.0.2-20171012160058+6cbbf247
    log:       /home/james/.cache/keybase/keybase.kbfs.log

Service:
    status:    running
    version:   1.0.34-20171012160058+c282fb227
    log:       /home/james/.cache/keybase/keybase.service.log

Updater:
    log:       /home/james/.cache/keybase/keybase.updater.log

Platform Information:
    OS:        linux
    OS vers:   Linux 3.13.0-129-generic x86_64-Ubuntu 14.04.5 LTS
    Runtime:   go1.9
    Arch:      amd64

Client:
    version:   1.0.34-20171012160058+c282fb227

Desktop app:
    status:    not running
    version:   
    log:       /home/james/.cache/keybase/Keybase.app.log

Config path:   /home/james/.config/keybase/config.json
Default user:  
Other users:   

command-line client:
    keybase status [pid: 6484, version: 1.0.34-20171012160058+c282fb227]
```

**Comment by strib** [comment]
@jamesongithub: we do not support multiple local users at once right now.  You'll need to run `run_keybase` as the user that actually needs keybase.  Sorry for the inconvenience.  (You should be able to install it from any user with sudo though.)

**Comment by jamesongithub** [comment]
@strib it seems to work on some other machines where I didn't use `run_keybase` though?
i installed it once with a user with sudo, then all users were able to use the client.

**Comment by jamesongithub** [comment]
this particular machine I used `run_keybase` and it's attached to the user somehow. is there any config I can change?

**Comment by strib** [comment]
`run_keybase` is required on any Linux box for KBFS/GUI to work.  Maybe you weren't using those features, only the `keybase` command line (which auto-forks a background service)?

Actually, I'm a little confused by your situation.  Is `james` the user with UID=1012, or not?

**Comment by jamesongithub** [comment]
yeah, only need the keybase cli for now. although good to know RE: kbfs/gui

yes. james is UID=1012.

**Comment by strib** [comment]
Oh, I guess you are running the `keybase` CLI from a non-`james` user, is that right?  Sorry, I didn't quite get it at first.

If so, that user definitely shouldn't be trying to access the keybased socket for `james`.  Does the non-`james` user have any "XDG" environment variables set, that might be pointing to the `james` runtime directory?

**Comment by jamesongithub** [comment]
yeah i just saw that now
i switched to another user to test.

`XDG_RUNTIME_DIR` was still set to my user.

what's the recommended way to work with this?

**Comment by jamesongithub** [comment]
one case where this failed was for jenkins build servers, some jobs run as another user to isolate priveleges

**Comment by strib** [comment]
We use that env var to figure out where sockets live.  Must be something strange going on in your configuration to set it by default for other users?  I guess either track that setting down and disable it, or override it for Keybase commands? I.e.:

```sh
XDG_RUNTIME_DIR="" keybase ...
```

**Comment by jamesongithub** [comment]
when you switch user with `su`, `XDG_RUNTIME_DIR` is maintained, it doesn't switch to the new user's UID i guess.

how does keybased run? seems like the debian install should create a keybase user and create 1 socket for everyone to use

I guess we can override XDG_RUNTIME_DIR for now. is there any others to know about?

**Comment by strib** [comment]
The socket gives free access to the secret key for whatever user is logged in, so we can't share a socket between users, it has to be access-controlled at the OS level.

We use all the XDG variables: "XDG_CONFIG_HOME", "XDG_CACHE_HOME", "XDG_DATA_HOME".  Or maybe you can try `su -` instead of plain `su`?

**Comment by jamesongithub** [comment]
i see.

yeah i tried that `su -` v `su` gives same environment variable for `XDG_RUNTIME_DIR`

thank you
